hazelcast/src/main/java/com/hazelcast/crdt/pncounter/PNCounterProxy.java

```diff
@@ -0,0 +1,319 @@
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public String getServiceName() {
+        return SERVICE_NAME;
+    }
+
+    @Override
+    public long get() {
+        return invoke(new GetOperation(name, observedClock));
+    }
+
+    @Override
+    public long getAndAdd(long delta) {
+        return invoke(new AddOperation(name, delta, true, observedClock));
+    }
+
+    @Override
+    public long addAndGet(long delta) {
+        return invoke(new AddOperation(name, delta, false, observedClock));
+    }
+
+    @Override
+    public long getAndSubtract(long delta) {
+        return invoke(new AddOperation(name, -delta, true, observedClock));
+    }
+
+    @Override
+    public long subtractAndGet(long delta) {
+        return invoke(new AddOperation(name, -delta, false, observedClock));
+    }
+
+    @Override
+    public long decrementAndGet() {
+        return invoke(new AddOperation(name, -1, false, observedClock));
+    }
+
+    @Override
+    public long incrementAndGet() {
+        return invoke(new AddOperation(name, 1, false, observedClock));
+    }
+
+    @Override
+    public long getAndDecrement() {
+        return invoke(new AddOperation(name, -1, true, observedClock));
+    }
+
+    @Override
+    public long getAndIncrement() {
+        return invoke(new AddOperation(name, 1, true, observedClock));
+    }
+
+    /**
+     * Invokes the {@code operation}, blocks and returns the result of the
+     * invocation.
+     * If the member on which this method is invoked is a lite member, the
+     * operation will be invoked on a cluster member. The cluster member is
+     * chosen randomly once the first operation is about to be invoked and if
+     * we detect that the previously chosen member is no longer a data member
+     * of the cluster.
+     * This method may throw a {@link com.hazelcast.crdt.ConsistencyLostException}
+     * if there are no reachable replicas which have observed the updates
+     * previously observed by users of this proxy. This means that the session
+     * guarantees are lost.
+     *
+     * @param operation the operation to invoke
+     * @return the result of the invocation
+     * @throws MutationDisallowedException                 if the cluster does not contain any data members
+     * @throws UnsupportedOperationException               if the cluster version is less than 3.10
+     * @throws com.hazelcast.crdt.ConsistencyLostException if the session guarantees
+     *                                                     have been lost
+     * @see ClusterService#getClusterVersion()
+     */
+    private long invoke(Operation operation) {
+        if (getNodeEngine().getClusterService().getClusterVersion().isLessThan(Versions.V3_10)) {
+            throw new UnsupportedOperationException(
+                    "PNCounter operations are not supported when cluster version is less than 3.10");
+        }
+        operation.setValidateTarget(false);
```

Comment:
afaik, target validation is only done for partition operations. We don't need to set this flag since we send operations to specific members here.

Reply:
Fixed.